src/consumption.cpp

```diff
@@ -759,7 +759,7 @@ bool player::eat( item &food, bool force )
 void cap_nutrition_thirst( player &p, int capacity, bool food, bool water )
 {
     if( ( food && p.get_hunger() < capacity ) ||
-        ( water && p.get_thirst() < capacity ) ) {
+        ( water && p.get_thirst() < capacity && !p.has_trait( trait_id( "NO_THIRST" ) ) ) ) {
```

Comment:
So you can overconsume liquid all you want?
That's a bad side effect of the change.  How about simply increasing hydration from food instead of having hydration be overridden?

Reply:
According to the research I did in the original PR, though mice are able to consume water and do so when they find it, they can also survive without it if they have sufficient nourishment as they extract the water they need from food. Staying true to that could be done a few ways.

Maybe make it so that eating food bumps their hydration up to a threshold if it's below that? Equalize the two like Mycus characters? Increasing the hydration of food would have to take a great deal of increasing to stay true to this.

Reply:
> Equalize the two like Mycus characters? 

This is closer to what I was thinking, I think it'll have fewer side effects.